File: business_rule_engine/Business Rule Engine/health-server.js
const express = require('express');
const cors = require('cors');

const app = express();
const PORT = process.env.PORT || 4003;

// Configure CORS with specific origins
app.use(cors({
    origin: [
        'http://localhost:3000',
        'http://localhost:4005',  // eCDome Intelligence
        'http://localhost:4006',  // Gamification
        'http://localhost:4007',  // Unified Integration
        'http://localhost:4008',  // Provider Dashboard
        'http://localhost:4009',  // Patient Dashboard
        'http://localhost:4011',  // Data Ingestion
        'http://localhost:4012',  // Biomarker GUI
        'http://localhost:8000',  // Telemedicine Platform
        'http://localhost:8080',  // API Gateway
    ],
    credentials: true,
    methods: ['GET', 'POST', 'PUT', 'DELETE', 'OPTIONS'],
    allowedHeaders: ['Content-Type', 'Authorization', 'X-Requested-With']
}));

app.use(express.json());

// Health check endpoint
app.get('/health', (req, res) => {
    res.json({
        status: 'healthy',
        module: 'business-rules',
        version: '1.0.0',
        timestamp: new Date().toISOString(),
        uptime: process.uptime()
    });
});

// Module info endpoint
app.get('/info', (req, res) => {
    res.json({
        id: 'business-rules',
        name: 'Business Rule Engine',
        version: '1.0.0',
        description: 'Conflict resolution and clinical decision support',
        endpoints: {
            health: '/health',
            info: '/info',
            api: '/api/v1'
        },
        dependencies: ['sdk-service'],
        capabilities: ['rule-management', 'conflict-resolution', 'decision-support']
    });
});

// Mock API endpoints
app.get('/api/v1/rules', (req, res) => {
    res.json({
        success: true,
        data: [
            { id: 'rule-001', name: 'Pain Management Rule', status: 'active' },
            { id: 'rule-002', name: 'Medication Interaction Rule', status: 'active' },
            { id: 'rule-003', name: 'Alert Threshold Rule', status: 'active' }
        ]
    });
});

app.post('/api/v1/conflicts/process', (req, res) => {
    res.json({
        success: true,
        data: {
            conflictId: 'conflict-001',
            resolution: 'automated-resolution',
            actions: ['alert-provider', 'update-patient-record'],
            timestamp: new Date().toISOString()
        }
    });
});

app.listen(PORT, () => {
    console.log(`Business Rules Health Server running on port ${PORT}`);
}); 

File: abena_ihr/health_server.py
from fastapi import FastAPI
import uvicorn
from datetime import datetime
import os

app = FastAPI()
PORT = int(os.getenv('PORT', 4002))

@app.get("/health")
async def health_check():
    return {
        "status": "healthy",
        "module": "abena-ihr",
        "version": "1.0.0",
        "timestamp": datetime.now().isoformat(),
        "database_connected": True
    }

@app.get("/info")
async def module_info():
    return {
        "id": "abena-ihr",
        "name": "Abena IHR Main System",
        "version": "1.0.0",
        "description": "Clinical outcomes management system",
        "endpoints": {
            "health": "/health",
            "info": "/info",
            "api": "/api/v1"
        },
        "dependencies": ["sdk-service"],
        "capabilities": ["patient-management", "outcome-tracking", "analytics"]
    }

@app.get("/api/v1/patients")
async def get_patients():
    return {
        "success": True,
        "data": [
            {"id": "patient-001", "name": "John Doe", "status": "active"},
            {"id": "patient-002", "name": "Jane Smith", "status": "active"},
            {"id": "patient-003", "name": "Bob Johnson", "status": "active"}
        ]
    }

@app.get("/api/v1/outcomes")
async def get_outcomes():
    return {
        "success": True,
        "data": [
            {"id": "outcome-001", "name": "Pain Reduction", "status": "tracking"},
            {"id": "outcome-002", "name": "Quality of Life", "status": "tracking"}
        ]
    }

if __name__ == "__main__":
    uvicorn.run(app, host="0.0.0.0", port=PORT) 